File: src/main/java/com/EmployeesREST/validation/CustomDateValidator.java
package com.EmployeesREST.validation;


import javax.validation.Constraint;
import javax.validation.Payload;
import java.lang.annotation.*;

@Documented
@Constraint(validatedBy = EmployeeDateOfBirthValidator.class)
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface CustomDateValidator {
    String message() default "Employee should be at least 18 y.o.";
    Class<?>[] groups() default {};
    Class<? extends Payload>[] payload() default {};
}

File: src/main/korlin/com/employees/validation/CustomDateValidator.kt
package com.employees.validation

import javax.validation.Constraint
import javax.validation.Payload
import kotlin.reflect.KClass

@MustBeDocumented
@Constraint(validatedBy = [EmployeeDateOfBirthValidator::class])
@Target(AnnotationTarget.FIELD)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
annotation class CustomDateValidator(
    val message: String = "Employee should be at least 18 y.o.",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = []
)